soroban-env-host/src/host.rs

```diff
@@ -2761,8 +2771,11 @@ impl VmCallerEnv for Host {
         let u = self.u8_from_u32val_input("u", u)?;
         let vnew = self.visit_obj(b, move |hv: &ScBytes| {
             self.validate_index_le_bound(i, hv.len())?;
-            let mut vnew: Vec<u8> = hv.metered_clone(&self.0.budget)?.into();
-            self.charge_budget(CostType::BytesInsert, hv.len() as u64)?; // insert is O(n) worst case
+            let mut vnew: Vec<u8> = hv.metered_clone(self.as_budget())?.into();
```

Comment:
here I think the same excessive charge is being applied?

Reply:
Fixed. See comments before.